packages/react-native/Libraries/AppDelegate/RCTDefaultReactNativeFactoryDelegate.mm

```diff
@@ -129,6 +129,10 @@ - (Class)getModuleClassFromName:(const char *)name
   return nullptr;
 }
 
+- (nonnull NSArray<NSString *> *)getModuleNames{
+   return [dependencyProvider moduleNames];
```

Comment:
we should check whether the dependencyProvider implement this method.
People could, in principle, create custom dependency providers